Userland/Libraries/LibWeb/Streams/AbstractOperations.cpp

```diff
@@ -2720,6 +2824,433 @@ WebIDL::ExceptionOr<void> writable_stream_default_controller_write(WritableStrea
     return {};
 }
 
+// https://streams.spec.whatwg.org/#initialize-transform-stream
+WebIDL::ExceptionOr<void> initialize_transform_stream(TransformStream& stream, JS::PromiseCapability& start_promise, double writable_high_water_mark, SizeAlgorithm&& writable_size_algorithm, double readable_high_water_mark, SizeAlgorithm&& readable_size_algorithm)
+{
+    auto& realm = stream.realm();
+
+    // 1. Let startAlgorithm be an algorithm that returns startPromise.
+    StartAlgorithm writable_start_algorithm = [&start_promise] {
```

Comment:
Is it safe to be capturing these by reference? I noticed that the caller initializes this on the stack, and we're passing through a reference (not a GC pointer)

Reply:
It's now a GC pointer